packages/expo-modules-core/src/ts-declarations/global.ts

```diff
@@ -31,6 +31,33 @@ export interface ExpoGlobal {
    */
   NativeModule: typeof NativeModule;
 
+  // Properties
+
+  /**
+   * The version of the `expo-modules-core` package.
+   */
+  expoModulesCoreVersion?: {
+    version: string;
+    major: number;
+    minor: number;
+    patch: number;
+  };
+
+  /**
+   * Boolean value indicating whether the new architecture is enabled.
+   */
+  isNewArchitectureEnabled?: string;
```

Comment:
```suggestion
  newArchEnabled?: string;
```

shorter and align the convention as react-native.
however, i think we don't actually like to introduce this property the these reasons:
- old arch would be deprecated soonish
- people now use `globalThis.RN$Bridgeless === true` to check the new architecture
- react-native core has some effort to recommend people remove build time conditions for new architecture. if means fabric or bridgeless code will also be built even on old arch. this already happens on core but not yet for 3rd party modules. i would like we remove the build time condition as well.
- as above, the build conditions will block prebuild effort that @lukmccall working on. we won't like to introduce `expo-modules-core-new-arch.aar` and `expo-modules-core-old-arch.aar`